app/app.rb

```diff
@@ -17,27 +17,29 @@ def initialize(socket_port: ENV.fetch("SOCKET_PORT"), socket_host: "0.0.0.0", so
     end
 
     def run
-      Hammeroids::Lobby.new.clear
-
       EventMachine.run do
 
         channel = EM::Channel.new
         EventMachine::WebSocket.start(host: @socket_host, port: @socket_port, secure: @socket_secure) do |connection|
-          player = Hammeroids::Player.new(connection, channel)
+          player = Hammeroids::Player.new
+
           connection.onopen do |handshake|
-            player.join
-            channel.push(Hammeroids::Lobby.new.to_json)
+            Hammeroids::Channels::Subscription.new(connection, channel).create
           end
 
           connection.onmessage do |message|
-            Hammeroids::Sockets::MessageRouter.new(connection, message).action
+            # TODO: refactor this once we have a better idea of all the events we'll be dealing with.
+            message_h = JSON.parse(message).deep_symbolize_keys
+            if message_h[:type] == "player"
+              player.update(message_h[:attributes])
+              Hammeroids::Channels::Lobby.new(channel).broadcast
+            end
             channel.push(message)
```

Comment:
If we get a message of type "player", the Lobby is broadcast, and calls channel.push with a json representation of itself. Then it looks like here the original message is broadcast as well. I guess it doesn't make much difference because we only process type 'lobby' in the frontend...

Reply:
Yeah I did think about putting an else around that second push, but I think we need to do more work on better message naming etc to identify which messages are for what. Also if i add an object to filter out the current connections data then i'll put that here.